src/mapml/layers/MapLayer.js

```diff
@@ -72,8 +82,88 @@ export var MapMLLayer = L.Layer.extend({
         this._container.style.opacity = opacity;
         if(this.opacityEl) this.opacityEl.value = opacity;
     },
+    _changeExtentOpacity: function(e){
+      if(e && e.target && e.target.value >=0 && e.target.value <= 1.0){
+        this.templatedLayer.changeOpacity(e.target.value);
+        this._templateVars.opacity = e.target.value;
+      }
+    },
+    _changeExtent: function(e, extentEl) {
+        if(e.target.checked){
+          extentEl.checked = true;
+          if(this._layerEl.checked){
+              extentEl.templatedLayer = M.templatedLayer(extentEl._templateVars, 
+                { pane: this._container,
+                  _leafletLayer: this,
+                  crs: extentEl.crs,
+                  extentZIndex: extentEl.extentZIndex
+                }).addTo(this._map);
+                extentEl.templatedLayer.setZIndex();
+                this._getCombinedExtentsLayerBounds();  
+            }       
+        } else {
+            L.DomEvent.stopPropagation(e);
+            extentEl.checked = false;
+            if(this._layerEl.checked) this._map.removeLayer(extentEl.templatedLayer);
+            this._getCombinedExtentsLayerBounds();
+        }
+    },
+
+    // if there are multiple extents, find the combined layerbounds
+    _getCombinedExtentsLayerBounds: function(){
```

Comment:
There should be only one way of setting `<layer->.extent`, regardless of how many `<map-extent>` elements there are.  This method would seem to be that way, need to factor out the other way.